### PR TITLE
Hometask day8

### PR DESCRIPTION
**Pull request by siarheibalashou**
added implementation of task 2,3,4,5,6,7
also .txt files from root directory added into separate folder for .gitignore

also added "spaces" for better readability

### CODE REVIEW

File: src/homework/day8/absclasses/collections/BirdsAsList.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+public class BirdsAsList {
+    public static void main(String[] args) {
+        List<String> birds = Arrays.asList("Чайка", "Дрозд", "Бусел", "Голубь", "Воробей", "Цапля");
+        for (String bird : birds) {
+            System.out.printf("-%s--", bird).println();
+        }
+        int counter = 0;
+
+        for (String bird : birds) {
+            if (bird.matches("[ауоиэы]{2,}")) {
```
Comment (Daria3495) [inline src/homework/day8/absclasses/collections/BirdsAsList.java]: Your regex doesn't work, it looks for 2 vowels in a row, that is why it print 0

File: src/homework/day8/absclasses/collections/BirdsAsList.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                counter++;
+            }
+        }
+        System.out.println("кол-во птиц с больше 1 гласной в имени: " + counter);
+        for (int index = 0; index < birds.size(); index++) {
+            System.out.print(birds.get(index) + " ");
+        }
+        System.out.println();
+        birds.set(3, "Синица");
```
Comment (Daria3495) [inline src/homework/day8/absclasses/collections/BirdsAsList.java]: index is incorrect, 4th element in the list is changed

File: src/homework/day8/absclasses/collections/ElementsAsArrayList.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            }
+
+        }
+        System.out.println();
+        System.out.println("кол-во элементов из больше двух слов: " + counter);
+
+        for (int index = 0; index < elements.size(); index++) {
+            System.out.print(elements.get(index) + " ");
+        }
+        elements.add(4, "Spinner");
```
Comment (Daria3495) [inline src/homework/day8/absclasses/collections/ElementsAsArrayList.java]: element is added not on 4th position, please review index

File: src/homework/day8/absclasses/collections/ElementsAsArrayList.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        elements.remove(2);
```
Comment (Daria3495) [inline src/homework/day8/absclasses/collections/ElementsAsArrayList.java]: the same problem as above - element which is 3rd in the list is removed

File: src/homework/day8/absclasses/collections/ElementsAsArrayList.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        elements.set(5, "Switch");
```
Comment (Daria3495) [inline src/homework/day8/absclasses/collections/ElementsAsArrayList.java]: index is incorrect

File: src/homework/day8/absclasses/collections/FiguresAsList.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+//Посчитать сколько фигур НЕ содержат букву "и" и вывести число в консоль
+//Проитерировать список по индексу и отпечатать слова в консоль через пробел
+//Вставить на 3 позицию Треугольник
+//Проитерировать список через for-each и отпечатать слова в консоль через пробел
+
+public class FiguresAsList {
+    public static void main(String[] args) {
+        List<String> figures = Arrays.asList("Овал", "Прямоугольник", "Круг", "Квадрат", "Эллипс");
+        try {
+            BufferedWriter writer = new BufferedWriter(new FileWriter("file_with_figures", true));
```
Comment (Daria3495) [inline src/homework/day8/absclasses/collections/FiguresAsList.java]: here I am not sure, should we each time add objects to file, in the task it looks like we should only add figures once, but if code is launched twice or more - file will be updated with new line of objects

File: src/homework/day8/absclasses/collections/FiguresAsList.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        for (String figure : figures) {
+            if (!figure.contains("и"))
+                counter++;
+        }
+        System.out.println("Кол-во слов без \"и\": " + counter);
+        for (int elemIndex = 0; elemIndex < figures.size(); elemIndex++) {
+            System.out.printf(" %s", figures.get(elemIndex));
+        }
+        System.out.println();
+        figures.set(3, "Треугольник");
```
Comment (Daria3495) [inline src/homework/day8/absclasses/collections/FiguresAsList.java]: Треугольник is on 4th position

**Review by Daria3495 — CHANGES_REQUESTED** [review]
Your code and realization are great, but honestly it's a bit hard to read it due to absence of spaces between cycles or other lines of code; would be great if you could separate the implementation of tasks using at least a space between appropriate line of codes for ease of reading

File: src/homework/day8/absclasses/collections/BirdsAsList.java (hunk 1 above)
Comment (siarheibalashou) [reply src/homework/day8/absclasses/collections/BirdsAsList.java]: fixed

File: src/homework/day8/absclasses/collections/BirdsAsList.java (hunk 2 above)
Comment (siarheibalashou) [reply src/homework/day8/absclasses/collections/BirdsAsList.java]: fixed

File: src/homework/day8/absclasses/collections/ElementsAsArrayList.java (hunk 3 above)
Comment (siarheibalashou) [reply src/homework/day8/absclasses/collections/ElementsAsArrayList.java]: fixed

File: src/homework/day8/absclasses/collections/ElementsAsArrayList.java (hunk 4 above)
Comment (siarheibalashou) [reply src/homework/day8/absclasses/collections/ElementsAsArrayList.java]: fixed

File: src/homework/day8/absclasses/collections/ElementsAsArrayList.java (hunk 5 above)
Comment (siarheibalashou) [reply src/homework/day8/absclasses/collections/ElementsAsArrayList.java]: fixed

File: src/homework/day8/absclasses/collections/FiguresAsList.java (hunk 6 above)
Comment (siarheibalashou) [reply src/homework/day8/absclasses/collections/FiguresAsList.java]: removed "true" in file writer, now file will be rewritten

File: src/homework/day8/absclasses/collections/FiguresAsList.java (hunk 7 above)
Comment (siarheibalashou) [reply src/homework/day8/absclasses/collections/FiguresAsList.java]: fixed

**Review by Daria3495 — APPROVED** [review]
Looks great, thank you
Only one note, as per my understanding each fix should be committed separately

**Pull request merged**